sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -73,6 +73,10 @@ class JsonFunctionsSuite extends QueryTest with SharedSQLContext {
     checkAnswer(
       df.select($"key", functions.json_tuple($"jstring", "f1", "f2", "f3", "f4", "f5")),
       expected)
+
+    checkAnswer(
+      df.selectExpr("key", "json_tuple(jstring, 'f1', 'f2', 'f3', 'f4', 'f5')"),
```

Comment:
@cloud-fan Thanks a lot for trying in hive. Wenchen, i searched for "lateral view" in spark code and didn't find a test case. I wanted to debug to study more about it. Also wenchen, i made a change in elementTypes computation like following 

override def elementTypes: Seq[(DataType, Boolean, String)] = fieldExpressions.zipWithIndex.map {
    case(l @ Literal(value, _), idx) if value.toString() != "null" =>
      (StringType, true, value.toString())
    case (_, idx) => (StringType, true, s"c$idx")
  }

I can now see the alias names correctly. I am not sure if this is the right change however. Do you have any thoughts ? Thank you.